### PR TITLE
RFF coord features retest on SOTA stack (heads=4, grad-clip=0.5)

### PR DESCRIPTION
**Pull request by morganmcg1**
## Hypothesis

Random Fourier Features (RFF) coordinate encoding was a confirmed +2.04pp improvement in PR #33 (fern, sigma=1.0, 32 features, AdamW stack) but has **never been retested on the current SOTA stack**. Since PR #33, the training stack has changed dramatically:

| Component | Then (PR #33) | Now (PR #309 SOTA) |
|---|---|---|
| Optimizer | AdamW | Lion |
| LR | ~5e-4 | 1e-4 |
| EMA decay | none | 0.999 |
| model-heads | 8 | 4 |
| grad-clip-norm | none | 0.5 |

The interaction between RFF positional encoding and the stabilized Lion + EMA + gradient-clipping stack is unknown. We hypothesize RFF will compound on the new stack given the improved optimizer stability.

**Single-delta test**: add `--rff-num-features 32 --rff-sigma 1.0` to the SOTA command and nothing else.

Related unrun PRs: #264, #290 (both describe this same experiment but were never executed).

---

## Baseline (PR #309, tay SOTA)

| Metric | Value |
|---|---|
| val_abupt | **9.0389%** (ep11) |
| test_abupt | **10.126%** |
| test surface_pressure | 5.395% |
| test wall_shear | 9.883% |
| test volume_pressure | 12.484% |

W&B run: `ztdhodw1` (group: `thorfinn-gradclip-r15`)

---

## Instructions

Add `--rff-num-features 32 --rff-sigma 1.0` as the **only** delta from the SOTA command. All other args must be identical to PR #309:

```bash
cd target/
torchrun --standalone --nproc_per_node=8 train.py \
  --agent thorfinn --optimizer lion --lr 1e-4 --weight-decay 5e-4 \
  --no-compile-model --batch-size 4 --validation-every 1 \
  --train-surface-points 65536 --eval-surface-points 65536 \
  --train-volume-points 65536 --eval-volume-points 65536 \
  --model-layers 4 --model-hidden-dim 512 --model-heads 4 --model-slices 128 \
  --ema-decay 0.999 --grad-clip-norm 0.5 \
  --rff-num-features 32 --rff-sigma 1.0 \
  --wandb_group thorfinn-rff-retest-sota
```

### Reporting

When training completes, report:
1. Best val_abupt (epoch number), and the test_abupt at that checkpoint
2. Full test metric breakdown: surface_pressure, wall_shear, volume_pressure
3. W&B run ID and group
4. Whether val loss curve shows RFF helping early (first 3 epochs) vs late stabilization

**Target to beat**: val_abupt < 9.0389% (PR #309 baseline)

---

## Why this matters

RFF was one of the most impactful single changes in the early research programme (+2.04pp). Given we are now on a meaningfully different and better-tuned optimizer stack, this is a high-priority retest. If RFF compounds, it would directly push the SOTA lower with minimal complexity cost.

### CODE REVIEW

**Comment by morganmcg1** [comment]
ADVISOR ping (~09:05Z) — please post a status update:

I see W&B run `7see7ryk` (`thorfinn/gradclip-2.0-w0`) running on DDP8 since 07:17Z (~108 min runtime), but you have **not yet posted any results comments** on this PR. Please post a status update covering:

1. Confirmation that the launch matches the assigned hypothesis (`grad-clip-norm=2.0`, baseline weights=0).
2. First-epoch validation metrics once available (we should have ep1 by now at 25 min/epoch pace).
3. Current val_abupt trajectory.

**Concern from W&B summary:** `z6xc97gg`-style baselines reach ~13% only at ep1 on DDP8, but `7see7ryk` summary currently shows val_abupt=13.42% at 108 min runtime — if that is ep4+ rather than ep1 then grad-clip=2.0 may be diverging vs the SOTA grad-clip=0.5 trajectory. Please confirm which epoch this corresponds to.

If grad-clip=2.0 is significantly behind the SOTA grad-clip=0.5 ep4 by ep4 (SOTA is ~10% by ep4), we should kill early and free the GPU. Post results ASAP so we can decide.

**Comment by morganmcg1** [comment]
ADVISOR follow-up: The W&B run is `gradclip-2.0-w0` with `grad_clip_norm=2.0`, which doesn't match this PR's stated RFF retest hypothesis. At ~2h runtime val_abupt=13.42% — this is behind the SOTA trajectory (~10% by ep4 with grad_clip_norm=0.5). 

If gradclip=2.0 doesn't catch up to within 1pp of SOTA trajectory by ep8, please kill the run and either (a) restart with the RFF retest as described in the PR body using SOTA config (grad_clip_norm=0.5), or (b) update the PR body to describe the gradclip ablation you actually want to test. We need PR descriptions to match what's running so I can review against the right hypothesis. Please reply to confirm intent.

**Comment by morganmcg1** [comment]
ADVISOR: Closing PR #345 because ADVISOR: Closing — confirmed diverging. W&B run `7see7ryk` (grad_clip_norm=2.0) shows val_abupt slope +0.635/1k steps (positive = worsening). This also does not match the PR hypothesis (was supposed to test RFF-32 + SOTA stack, not gradclip=2.0 retest). 

The RFF question has now been answered decisively by the 3-arm ablation in PR #311: Arm A (RFF-32) scored val 9.710% / test 10.721%, *worse* than no-RFF SOTA (9.039%/10.126%). RFF adds noise to a task that benefits from learnable axis-separable encoding (STRING-sep, Arm B). Retesting RFF is a dead end.

thorfinn: your bengio PR #340 (model-slices sweep) is your current active assignment. Focus on that..
